src/snowflake/snowpark/session.py

```diff
@@ -974,12 +1012,143 @@ def get_req_identifiers_list(
 
             return res
 
-        # always include cloudpickle
+        # Add dependency packages
+        if dependency_packages:
+            for package in dependency_packages:
+                name = package.name
+                version = package.specs[0][1] if package.specs else None
+
+                # Add to packages dictionary
+                if name in result_dict:
+                    if result_dict[name] != str(package):
+                        raise ValueError(
+                            f"Cannot add dependency package '{name}'{'( version '+version+')' if version else ''} "
+                            f"because {result_dict[name]} is already added."
+                        )
+                else:
+                    result_dict[name] = str(package)
+
+        # Add the Snowpark package to packages dict, based on client's environment (or latest), if not already added
+        add_snowpark_package(result_dict, valid_packages)
```

Comment:
Curious: why do we always need to add snowpark packages?

Reply:
It's a convenience for sprocs, not having to specify snowpark; maybe this can be added at the  sproc registration point instead so as to not affect UDFs.

Reply:
Removed these changes for now.